Add new Bitget connector

gtwototoo opened the pull request:
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] Tests all pass
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
1. Added a new Bitget spot connector
2. Added tests for the Bitget spot connector
3. Updated the Bitget perpetual connector from version 1 to version 2
4. Updated Bitget perpetual connector tests

**Tests performed by the developer**:



**Tips for QA testing**:

gtwototoo commented:
> Please advise when the PR is ready for review @gtwototoo thank you

@rapcmia Good evening, PR ready for review

rapcmia commented:
> @rapcmia Good evening, PR ready for review

Thanks for the update, will start checking on it today 🙏🏼

rapcmia commented:
Hi @gtwototoo the github checks failed
```

=========================== short test summary info ============================
FAILED test/hummingbot/connector/derivative/bitget_perpetual/test_bitget_perpetual_order_book_data_source.py::BitgetPerpetualAPIOrderBookDataSourceTests::test_get_funding_info - KeyError: 0
FAILED test/hummingbot/connector/derivative/bitget_perpetual/test_bitget_perpetual_user_stream_data_source.py::BitgetPerpetualUserStreamDataSourceTests::test_listening_process_authenticates_and_subscribes_to_events - AssertionError: {'op'[71 chars]t', 'instId': 'default'}, {'instType': 'USDT-F[122 chars]t'}]} != {'op'[71 chars]t', 'coin': 'default'}, {'instType': 'USDT-FUT[116 chars]t'}]}
- {'args': [{'channel': 'account',
+ {'args': [{'channel': 'account', 'coin': 'default', 'instType': 'USDT-FUTURES'},
+           {'channel': 'positions',
-            'instId': 'default',
?               ----

+            'coin': 'default',
?             ++

             'instType': 'USDT-FUTURES'},
+           {'channel': 'orders', 'coin': 'default', 'instType': 'USDT-FUTURES'}],
-           {'channel': 'positions',
-            'instId': 'default',
-            'instType': 'USDT-FUTURES'},
-           {'channel': 'orders',
-            'instId': 'default',
-            'instType': 'USDT-FUTURES'}],
   'op': 'subscribe'}
===== 2 failed, 4827 passed, 1 skipped, 215 warnings in 313.63s (0:05:13) ======
make: *** [Makefile:14: test] Error 1
Error: Process completed with exit code 2.
0s

```

gtwototoo commented:
@rapcmia Check again please

nikspz commented:
commit 47fd19e2da0ac19bc4cad6e4b3a8c7b96e410cac

- Bitget Perpetual
    - connected successfully
    - added USDC for USDC-M Futures
    - added SOL for Coin-M Futures
    - balance: ok, showing USDT,USDC,SOL
    - created perpetualMM using bitget_perpetual
    - markets:
        - USDT-M Futures: ok
            - now showing 1 - USDT pairs
        - USDC-M Futures: ok
        - Coin-M Futures: ok
    - leverage: ok
    - started successfully
    - status —live : ok
    - opened position successfully
    - closed with take profit successfully
    - opened position successfully
    - closed with stop-loss successfully
    - position retrieval: ok
    - history: ok
    - fees: ok
- pending: longrun test

gtwototoo commented:
@rapcmia Please test the spot again. I found different logic when creating MARKET BUY orders. I fixed the issue and changed the calculations for order fills.

rapcmia commented:
Check bot running for 12hours 
<img width="903" height="71" alt="image" src="https://github.com/user-attachments/assets/1abfafc0-7e62-4cfc-af1f-8e5acda3ca82" />
```
####### EXECUTOR TABLE SUMMARY #######
Distinct Close Types:  3
- Time limit (1): 22
- Stop loss (2): 60
- Take profit (3): 3

####### TRADEFILL TIMING & SUMMARY #######
Start Time:            2025-10-02 14:21
End Time:              2025-10-03 03:03
Total Runtime:         0d 12h 41m
Total Trades:          186
Total Volume (quote):  1252.144128
Buy Orders:            98
Sell Orders:           88
Trade PnL (quote):     -7.889566
```
  - Review log file
      - Orders are placed, cancelled and filled ok
      - No error found nor hit rate limit
      - Compare trade data between CSV and exchange order history, all matched ✅

rapcmia reviewed (approved):
QA approved 1/2

- Tested using source build on linux
- Tested on spot only
- Created lib locally and added API keys on hummingbot-api ✅
- Compare data between client and exchange: Balance and Orderbook ✅
- Setup v2 market_making.pmm_simple and generic.xemm_multiple_levels
	- Orders matched between client adn exchange 
	- limit/market order supported (tested on stop-loss)
	- limit-maker not supported ❌
	- Build docker image then tests with xemm controller ✅
- Dev fix reported issue on tests

cardosofede reviewed (approved):
great work @gtwototoo !
